### PR TITLE
chore: update or pin action versions, workflow naming and action's config

### PR DESCRIPTION
**Pull request by pmoscode**
## Description

This is a complete check of all GH workflows. What was checked and updated:
- Workflow naming 
- action's version (and config when required)
- workflow's agent
- trigger branch names

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Review by the-tatanka — APPROVED** [review]
LGTM

File: .github/workflows/veracode.yaml [hunk 1]
```diff
@@ -55,17 +55,18 @@ jobs:
...
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
-            java-version: 17
+          distribution: 'temurin'
+          java-version: '17'
 
       - name: Build with Gradle
-        uses: gradle/gradle-build-action@4137be6a8bf7d7133955359dbd952c0ca73b1021
+        uses: gradle/gradle-build-action@v2Ï
```
Comment (carslen) [inline .github/workflows/veracode.yaml:64]: This version (`v2Ï`) looks like a typo.

**Review by carslen — CHANGES_REQUESTED** [review]

File: .github/workflows/veracode.yaml (hunk 1 above)
Comment (pmoscode) [reply .github/workflows/veracode.yaml]: Yep, was a typo...

**Review by carslen — APPROVED** [review]
LGTM

**Pull request merged**